### DISCUSSION TITLE
Issue with Redirect URL for MS AD Auth, going to http instead of given https

### DISCUSSION THREAD

**Discussion by AgioGlobal-IT** [datalust/seq-tickets#1912] [Help]
Hi,
I have encountered an issue. I have HAproxy to terminate my ssl and connect it to an seq docker container running on port 80, nothing special.
I enabled the Azure AD connector, setup the app in AD en all looks OK.
When trying to logon, the url send to MS for the callback gets "http://my-external-url" while this should be "https://my-external-url" (note http vs https)
Even set in the config where you have the option: "Redirect URI" this entry contains the HTTPS version.
tried as well X-Forward headers in haproxy  .. but nothing fixes the url ..
MS auth screen mentions all the time that the http url is given along.
Trying to curcomvent this in the app settings from MS is not working, since MS does not accept an callback url which is http.
So I am little stuck.
I searched and did not find simular reports, am I the first .. overlooked something ??
Any help even a param that would force the callback url given in the initial request to MS would help, but I also did not found any option for such a config.
Any help suggestions are welcome.
Regards
Kees

**Comment by nblumhardt** [comment]
Hi! You need to set:

```
SEQ_API_CANONICALURI=https://my-external-url
```

as an environment variable in the Seq container. Hope this helps.

**Comment by AgioGlobal-IT** [comment]
Hi,
Thanks, this worked.
I found this somewhat confusing from the Documentation && Screens of the application itself.

Regards,
Kees

________________________________

**Reply by nblumhardt** [reply]
Thanks for the feedback. I've added a reference to `SEQ_API_CANONICALURI` to the README on `hub.docker.com`; is there another location in particular where you think it would be worthwhile adding more about this? Thanks again!

**Comment by AgioGlobal-IT** [comment]
Hi Nicolas,

Well I was just following this doc: https://docs.datalust.co/docs/azure-active-directory

Half way it mentions this section:



[cid:85954dde-0c1a-4156-98ae-68999243f3b5]

So first of all you don't get to see the config of the CallBackURL.

So my first suggestion would be to show that part as well

Secondly I would suggest putting a little attention block above it like this:
[cid:e871d14d-1be7-4608-8157-e7308501be94]
Stating the message and or link:

If you run seq in a docker container, dont forget to enable / add the following environment variable: SEQ_API_CANONICALURI=YourUrl

And also explain that although you see it there in the configuration screen, this does not work as expected and is mandatory ..

Hope it helps,
Regards,



________________________________

**Reply by nblumhardt** [reply]
Thanks! I've updated the text in the first (blue) suggestion box, and added a new Troubleshooting section to the end, which should improve things 👍

**Comment by AgioGlobal-IT** [comment]
Hi Nicholas,
I looked at what you mentioned.

I dont want to be a nuisance, but mentioning the : "Windows" threw me off immediately, meaning to say, I stop reading there.

Reason is I run the instance in a docker container on Linux.

Why not ad a more generic section mentioning, if you run the instance in a docker
followed by
on Windows do this ..
on Linux do that ..

Just a humble suggestion.
Regards,
Kees


________________________________

**Reply by nblumhardt** [reply]
That's a great point; thanks. I've made a couple of tactical `(Docker)` additions, but I'll run back over this when I have a chance 👍
